dist/iconpicker-1.5.0.js

```diff
@@ -1,439 +1,432 @@
-                    singleIconElm.addEventListener('click', function (e) {
-                        e.preventDefault();
-                        var iconClassName = this.dataset.class;
-
-                        // each input value on
-                        for (var i = 0; i < inputElm.length; i++) {
-
-                            var getTagName = inputElm[i].tagName.toString().toLowerCase();
-                            // if an input or textarea element
-                            if (getTagName === 'input' || getTagName === 'textarea') {
-                                inputElm[i].value = iconClassName;
-                            }
-                            // else any of element
-                            else {
-                                inputElm[i].innerHTML = iconClassName;
-                            }
-
-                        }
-                        // each input value off
-
-                        // each preview on
-                        for (var i = 0; i < previewElm.length; i++) {
-                            previewElm[i].className = iconClassName;
-                        }
-                        // each preview off
-
-                        // callback on
-                        if (theCallback) {
-                            theCallback();
-                        }
-                        // callback off
-
-                        removeIpElement(400);
-                    }, false);
-                }
-                // add listeners each off
+            var noResultElm = '<p class="ip-no-results-found">' + noResultsText + '</p>';
+            tempResults.innerHTML = noResultElm;
+          }
+          // no results found off
 
-            }
-            // each icon click listener off
+          // append temp container to results area
+          searchResultArea.appendChild(tempResults);
 
-            // first icons listeners
-            eachIconEventListener('first');
+          // event listener for each temp icon
+          eachIconEventListener('search');
 
         }
-        // IconPicker: Append Library to Body off
-    },
+        // show first icons
+        else {
+          firstIconsArea.style.display = 'block';
+        }
+
+      }, false);
+      // search input keyup listener off
+
+
+      // each icon click listener on
+      function eachIconEventListener(firstOrSearch) {
+
+        var inputElm = document.querySelectorAll(inputElement);
+
+        // define icons on
+        var eachIconElm;
+        if (firstOrSearch === 'first') { // first
+          eachIconElm = document.getElementById(IconPickerModal.id).getElementsByClassName('first-icon');
+        } else if (firstOrSearch === 'search') { // search
+          eachIconElm = document.getElementById(IconPickerModal.id).getElementsByClassName('search-icon');
+        }
+        // define icons off
+
+        // add listeners each on
+        var wrapElem = document.querySelector('.ip-icons-content');
```

Comment:
I would add small note why the observer:
`                // this is required due to case where we use the js based fontawesome as fontawesome changes <i> to <svg>`